website/blog/2024-08-13-distributed-execution-flow-paradigms/index.mdx

```diff
@@ -0,0 +1,335 @@
+
+Durable execution isn’t a completely distinct solution from the ones listed above but rather incorporates some of their strengths while presenting a more comprehensive approach to achieving consistency, fault tolerance, data integrity, resilience for long-running processes, and observability.
+
+<img
+  src="/images/blog/execution-flow-paradigms/durable-exec-temporal.png"
+  alt="Durable workflow engine - Temporal"
+/>
+<div style={{ marginLeft: 10 + "em" }}>
+  Fig. [Durable workflow engine -
+  Temporal](https://temporal.io/blog/workflow-engine-principles)
+</div>
+
+#### Advantages of Durable Execution
+
+**Reduced Manual Intervention**: Minimizes the need for human intervention by handling retries and failures programmatically.
+**Improved Observability**: Provides a clear audit trail and visibility into the state of workflows, which aids in debugging and monitoring.
+**Scalability**: Scales efficiently across distributed systems while maintaining workflow integrity.
+
+#### Challenges of Durable Execution
+
+**Resource Intensive**: Persistent state storage and management can consume significant resources, especially in large-scale systems.
+**Latency**: The need to persist state and handle retries can introduce latency in the execution flow.
+
+As durable execution grows to be a fundamental driver of distributed computing, some of the solutions which use this architecture are
+
+- [Temporal](https://temporal.io/)
+- [Azure Durable Functions](https://learn.microsoft.com/en-us/azure/azure-functions/durable/)
+- [Amazon SWF](https://docs.aws.amazon.com/swf/)
+- [Uber Cadence](https://cadenceworkflow.io/)
+- [Infinitic](https://www.infinitic.io/)
+
+Among these, [Temporal](https://temporal.io/) has grown in influence, used by companies like SnapChat, HashiCorp, Stripe, DoorDash, and DataDog. Its success is driven by its practical application in real-world scenarios and the expertise of its founders.
+
+At Metatype, we recognize the value of durable execution and are committed to making it accessible. Our [Temporal Runtime](/docs/reference/runtimes/temporal) integrates seamlessly into our declarative API development platform, enabling users to harness the power of Temporal directly within Metatype. For those interested in exploring further, our documentation provides a detailed guide on getting started with [Temporal Runtime](/docs/reference/runtimes/temporal).
+
+Below is an example of how you can build a simple API to interact with an order delivery temporal workflow within Metatype.
+
+:::note
+If you are new to Metatype or haven’t set it up yet in your development environment. You can follow this [guideline](/docs/tutorials/quick-start).
+:::
+
+For this example, the order delivery system will have few components/services such as Payment, Inventory and Delivery.
+
+Your temporal workflow definition should look similar to the one below.
```

Comment:
Ending with a large code-dump is highly unreadable. I suggest you break it down and intersperse it with explanations of what each section is trying to accomplish. You also don't really need to showcase it in two languages.